salt/states/boto_iam.py

```diff
@@ -472,16 +476,32 @@ def group_present(name, policies=None, policies_from_pillars=None, users=None, r
     if not _ret['result']:
         ret['result'] = _ret['result']
         return ret
-    # TODO: make this work like _group_policies_present. This is currently not
-    #       fully idempotent.
     if users:
         log.debug('Users are : {0}.'.format(users))
         for user in users:
             log.debug('User is : {0}.'.format(user))
             ret = _case_group(ret, user, name, region, key, keyid, profile)
+        for user in _case_delete(users, exists):
```

Comment:
What I mean is that _case_group should ensure the state of something, rather than just adding a user to a group. No matter where you're calling these functions, the behavior should be the same (it should be idempotent). If you're adding users to a group in one spot, you should also be ensuring users that aren't listed are being removed too.